src/components/badge/Badge.tsx

```diff
@@ -0,0 +1,48 @@
+import { Component, Show } from 'solid-js'
+import type { JSX } from 'solid-js'
+
+import StyledContainer from './StyledContainer'
+import StyledIcon from './StyledIcon'
+
+export type Color =
+  | 'default'
+  | 'grey'
+  | 'purple'
+  | 'green'
+  | 'teal'
+  | 'primary'
+  | 'secondary'
+  | 'danger'
+
+export type BadgePlacement =
+  | 'top-left'
+  | 'top-right'
+  | 'bottom-left'
+  | 'bottom-right'
+
+export type BadgeProps = {
+  count?: number
+  countLimit?: number
+  color?: Color
+  children?: JSX.Element
+  badgeColor?: Color
+  badgePlacement?: BadgePlacement
+  offsetX?: number
+  offsetY?: number
+}
+
+const Badge: Component<BadgeProps> = (props) => {
+  let countLimit = props.countLimit || 99
```

Comment:
I don't think this would be reactive. If the `props.countLimit` changes after initial mount, it may not update right?

I made a util called `withDefaults` which can be used to specify default props in a reactive way, maybe you could utilise this? The button component uses it currently.